### ISSUE TITLE
References to root path in newData context should use parent() traversal.

### ISSUE THREAD

**Issue by mckoss** [firebase/bolt#59]
Since we don't have a newRoot variable in Firebase rules, references that access the root (w/o a prior wrapper) should use the newData.parent() trick to reference a relative path.  For example:
```
path /secret/$uid is String {
  read() = isUser($uid);
  validate() = root.users[$uid] != null;
}

path /users/$uid is User {
  read() = true;
  validate() = root.secret[$uid] != null;
}
```

**Comment by mckoss** [comment]
Resolved by f0b108fcdae66ffdee82f6e01b77589d59ddc572

**Comment by katowulf** [comment]
☃
